pkgs/by-name/sh/shell-gpt/package.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  python3,
+  fetchFromGitHub,
+  nix-update-script,
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "shell-gpt";
+  version = "1.4.3";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "TheR1D";
+    repo = "shell_gpt";
+    rev = "refs/tags/${version}";
+    hash = "sha256-T37L4U1kOrrIQJ2znq2UupD3pyit9xd8rAsEwUvGiQ8=";
+  };
+
+  pythonRelaxDeps = [
+    "requests"
+    "rich"
+    "distro"
+    "typer"
+    "instructor"
+  ];
+
+  build-system = with python3.pkgs; [ hatchling ];
+
+  nativeBuildInputs = with python3.pkgs; [ pythonRelaxDepsHook ];
+
+  dependencies = with python3.pkgs; [
+    markdown-it-py
+    rich
+    distro
+    typer
+    requests
+    hatchling
+    openai
+    instructor
```

Comment:
Do we really need `markdown-it-py`, `requests` and `hatchling` at runtime?
Also, we should add `click` to dependencies.

https://github.com/TheR1D/shell_gpt/blob/6f9d7f6fd3b99c60bcc7965222d837bb3d43c400/pyproject.toml#L31-L37

Reply:
Thanks, cleaned.